media-api/app/controllers/MediaApi.scala

```diff
@@ -89,12 +89,13 @@ class MediaApi(
   private def isUploaderOrHasPermission(
     principal: Principal,
     image: Image,
-    permission: SimplePermission
+    permission: SimplePermission,
+    uploaderPermission: SimplePermission
   ) = {
     principal match {
       case user: UserPrincipal =>
         if (user.email.toLowerCase == image.uploadedBy.toLowerCase) {
-          true
+          authorisation.hasPermissionTo(uploaderPermission)(principal)
         } else {
           authorisation.hasPermissionTo(permission)(principal)
         }
```

Comment:
The logic here doesn't let someone who has `permission` but not `uploaderPermission` take the action on their own images. In reality anyone with `permission` will probably have `uploaderPermission` but it's a weird corner case.
This is likely better as `(isUploader && hasUploaderPermission) || hasPermission` to deal with that case rather than an if expression.